### DISCUSSION TITLE
Review of the past year in your company.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#245] [General]
It has been a year of sharing and exchanges. You have developed BiblioteQ by adding a lot of features and answering my requests for evolutions.

I am very grateful for your patience and your explanations (sometimes too technical for my modest level) but which have the merit of enlightening the features developed in a constructive spirit and to improve BQ.

I come back to my role as a BiblioteQ user and the difficulty to make a clear request for the evolution of the software (it's a job to define the users' needs). 
Yes, it is also a difficult process because I am not a developer, I don't know if my requests are difficult, or even impossible to do; or on the contrary, if they can be done in 5 minutes: you have already done it many times.

My metaphor Mountain/Hill and development time suits me well.

I hope I have been helpful in giving feedback as a brand new user of BiblioteQ.

I had some good surprises and some requests that unfortunately fell in the TODO list.
![BQ-TOLIST](https://user-images.githubusercontent.com/56431039/197395278-7ded4e04-dfbe-4177-93f5-507d22ad3fd8.jpg)

I am attaching a file that lists the year's improvements and looking back, I see the impressive number of topics covered. BRAVO for all these evolutions.
[BQ_list_of_improvements.pdf](https://github.com/textbrowser/biblioteq/files/9846701/BQ_list_of_improvements.pdf)

I don't know yet which software will be used by the volunteers of my library (decision in November) but I had a great year with you and BiblioteQ.
![BQ-Relax](https://user-images.githubusercontent.com/56431039/197395345-d2a7a3b5-b474-4a15-912f-d2d395dfd312.jpg)

Good mood, exchanges, American point of view, textbrowser point of view, English language : it has been very pleasant and entertaining.
I hope that you will reserve many more evolutions of BiblioteQ and I will not miss to follow the 2nd season. I know where it happens :-)

![BQ-Avenir](https://user-images.githubusercontent.com/56431039/197395375-6e3cac90-7cb3-4e82-ae1b-9866b5800362.jpg)

I will probably be less active in requests because the main ones are now almost all in TODO list. But I will not fail to answer you in case of requests for clarifications.

In conclusion: 
From my point of view, you share with BiblioteQ the best software in its category so THANK YOU again for your dedication to free software and volunteer librarians.

Good Continuation

**Comment by textbrowser** [comment]
Which path is best.

You state things like "I want to be able to see window A, B, and C" in my resolution. I respond with an image depicting my work environment which should persuade you that your problem is not a problem. I ask the opposite question. Can I have an option to request from the underlying display manager to place a window on-top with respect to other windows so that when I click on another window my important window is not hidden? Now this offers new creativity and problems. As a solution, it solves a background / foreground preference. As a problem, it doesn't solve Android  and it requires that the application record the preference. This is the detailed understanding of an otherwise simple problem.

Statistics. A correct solution requires 6 months of total time because it has to be customizable. It has to present valuable mechanisms for gathering data and displaying it in your view. I don't have an interest in this because I can use other tools if I need to do this. So, I can spend 6 months for something that you will eventually drop and which will eventually never be of benefit to anyone else. It's like me going outside, painting a Michelangelo and burying it in the dirt. Lots of work, lots of glamour, lots of utility to be forgotten.

Batch Activities. The process is simple. It was at least. I proposed a scan and finalize approach. It's now a bunch of mixtures which confuses me more but apparently works for you.

Etc.

I prefer generic solutions and if custom solutions are absolutely necessary than the application has to grow to become generic for those custom solutions: that is, customizable.

Either or thinking shackles. Indecision is a problem.

My suggestion: I would like A. Describe A and limit A's scope. Once A is complete, enhance A. Batch Activities: 1. Scan item A. 2. Create a row. 3. Fill row with scanned text. To me that's a completion. Can it be enhanced? Yup. Small adjustments. If you go too far either way, you fall off the cliff.

**Comment by textbrowser** [comment]
You saw the amount of work required for formatting ISBNs. That's a custom solution which is expensive.

**Reply by meteos77** [reply]
if I had known that it would cause too much complication for future development I would have done without it.
it's more pleasant to read barcodes with dashes

I can't foresee in advance the consequences of my request?

**Reply by textbrowser** [reply]
It's not a regret.

**Comment by textbrowser** [comment]
Success is not about eliminating paradoxes. It's learning to live with them. User-friendliness is a paradox that cannot be eliminated. Like, a bike is not user friendly. With practice, it does become user friendly.

**Comment by meteos77** [comment]
I will try to be better the second year :-)
